src/views/Package/PackageHeader/SelectVersion.tsx

```diff
@@ -0,0 +1,65 @@
+import { Box, Text, Select } from "@chakra-ui/react";
+import { FunctionComponent } from "react";
+import { useHistory } from "react-router-dom";
+import { useCatalogResults } from "../../../hooks/useCatalogResults";
+import { getPackagePath } from "../../../util/url";
+import { usePackageState } from "../PackageState";
+
+export const SelectVersion: FunctionComponent = () => {
+  const { scope, name, version, language } = usePackageState();
+  const pkgName = scope ? `${scope}/${name}` : name;
+  const packages = useCatalogResults({
+    query: pkgName,
+    exactMatch: true,
+    limit: 20,
+  });
+  const { push } = useHistory();
+
+  const packageMajorVersions = packages.results.map((pkg) => ({
+    major: pkg.major,
+    version: pkg.version,
+  }));
+  packageMajorVersions.sort((a, b) => a.major - b.major);
+  const defaultMajor = packageMajorVersions.find(
+    (mv) => mv.version === version
+  );
```

Comment:
⚠️ I would compute this in a single `useMemo` effect. The current implementation means that `packageMajorVersions` will be recomputed on every render

Reply:
Updated.